platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py

```diff
@@ -785,6 +791,77 @@ def get_tx_fault(self):
         api = self.get_xcvr_api()
         return [False] * api.NUM_CHANNELS if api else None
 
+    def get_temperature(self):
```

Comment:
@Junchao-Mellanox why we need this thread to read the optics temperature when the same is available in TRANSCEIVER_DOM_SENSOR table

Reply:
Hi Prince, thanks for the comments. DOM info is updated every 1 minutes. However, this get_temperature function might be called at any time. So, we decided to have implementation not from DB.